src/types.ts

```diff
@@ -218,12 +219,14 @@ export interface SourceComparison {
     expectedSlippage: BigNumber | null;
 }
 
-export interface AffiliateFee {
-    feeType: AffiliateFeeType;
-    recipient: string;
-    sellTokenPercentageFee: number;
-    buyTokenPercentageFee: number;
-}
+export type AffiliateFee =
+    | { feeType: AffiliateFeeType.GaslessFee; recipient: string }
+    | {
+          feeType: AffiliateFeeType.None | AffiliateFeeType.PercentageFee | AffiliateFeeType.PositiveSlippageFee;
```

Comment:
We should split out `PositiveSlippageFee` as it doesn't make use of `sellTokenPercentageFee` or `buyTokenPercentageFee`.

Apart from that, LGTM